src/main/kotlin/ecommerce/dto/product/ProductRequest.kt

```diff
@@ -0,0 +1,41 @@
+package ecommerce.dto.product
+
+import ecommerce.validation.NAME_LENGTH_MAXIMUM
+import ecommerce.validation.PRODUCT_NAME_PATTERN
+import ecommerce.validation.PRODUCT_PRICE_MINIMUM
+import ecommerce.validation.URL_PATTERN
+import ecommerce.validation.UniqueProductName
+import jakarta.validation.constraints.DecimalMin
+import jakarta.validation.constraints.NotBlank
+import jakarta.validation.constraints.Pattern
+import jakarta.validation.constraints.Size
+
+data class ProductRequest(
+    @field:Size(max = NAME_LENGTH_MAXIMUM, message = ErrorMessages.NAME_SIZE)
+    @field:NotBlank(message = ErrorMessages.NAME_BLANK)
+    @field:Pattern(
+        regexp = PRODUCT_NAME_PATTERN,
+        message = ErrorMessages.NAME_PATTERN,
+    )
+    @field:UniqueProductName
```

Comment:
Wow, that’s impressive! Just be mindful that this query is always being executed.